plugins/jobs/web_client/JobStatus.js

```diff
@@ -47,12 +53,20 @@ var JobStatus = {
     registerStatus: function (status) {
         _.each(status, function (info, name) {
             this[name] = info.value;
-            this._map[info.value] = {
+
+            let statusInfo = {
                 text: info.text,
                 icon: info.icon,
                 color: info.color,
+                textColor: 'white',
                 finished: _.has(info, 'finished') ? info.finished : false
             };
+
+            if ('textColor' in info) {
```

Comment:
`in` is a bit tricky to use, since it checks up the entire prototype chain. Maybe you want [`_.has`](http://underscorejs.org/#has)?